uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -293,75 +114,68 @@ define([
-     */
-    function loadPublicHolidays() {
-      return PublicHoliday.all()
-        .then(function (publicHolidaysData) {
-          var datesObj = {};
-
-          // convert to an object with time stamp as key
-          publicHolidaysData.forEach(function (publicHoliday) {
-            datesObj[getDateObjectWithFormat(publicHoliday.date).valueOf()] = publicHoliday;
-          });
-
-          publicHolidays = datesObj;
-        });
-    }
-
-    /**
-     * Loads the status option values
-     *
-     * @return {Promise}
-     */
-    function loadStatuses() {
-      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_status')
-        .then(function (statuses) {
-          leaveRequestStatuses = _.indexBy(statuses, 'value');
-        });
-    }
+        return vm._loadCalendar();
+      })
+      .then(function () {
+        vm._showMonthLoader();
+      });
+    };
 
     /**
      * Sets UI related properties(isWeekend, isNonWorkingDay etc)
      * to the calendar data
      *
      * @param  {object} calendar
-     * @return {object}
      */
-    function setCalendarProps(calendar) {
-      var dateObj,
-        leaveRequest,
-        dates = Object.keys(calendar.days);
+    vm._setCalendarProps = function (calendar) {
+      var leaveRequest,
+        monthData = _.clone(vm.months);
 
-      dates.forEach(function (date) {
-        dateObj = calendar.days[date];
-        leaveRequest = getLeaveRequestByDate(dateObj.date);
+      _.each(calendar.days, function (dateObj) {
+        //fetch leave request, search by date
+        leaveRequest = vm.leaveRequests[dateObj.date];
 
         dateObj.UI = {
-          isWeekend: calendar.isWeekend(getDateObjectWithFormat(dateObj.date)),
-          isNonWorkingDay: calendar.isNonWorkingDay(getDateObjectWithFormat(dateObj.date)),
+          isWeekend: calendar.isWeekend(vm._getDateObjectWithFormat(dateObj.date)),
+          isNonWorkingDay: calendar.isNonWorkingDay(vm._getDateObjectWithFormat(dateObj.date)),
           isPublicHoliday: vm.isPublicHoliday(dateObj.date)
         };
 
         // set below props only if leaveRequest is found
         if (leaveRequest) {
-          dateObj.UI.styles = getStyles(leaveRequest, dateObj);
-          dateObj.UI.isRequested = isPendingApproval(leaveRequest);
-          dateObj.UI.isAM = isDayType('half_day_am', leaveRequest, dateObj.date);
-          dateObj.UI.isPM = isDayType('half_day_pm', leaveRequest, dateObj.date);
+          dateObj.UI.styles = vm._getStyles(leaveRequest, dateObj);
+          dateObj.UI.isRequested = vm._isPendingApproval(leaveRequest);
+          dateObj.UI.isAM = vm._isDayType('half_day_am', leaveRequest, dateObj.date);
+          dateObj.UI.isPM = vm._isDayType('half_day_pm', leaveRequest, dateObj.date);
         }
+
+        vm._getMonthObjectByDate(moment(dateObj.date), monthData).data.push(dateObj);
       });
 
-      return calendar;
-    }
+      vm.months = monthData;
+    };
+
+    (function init() {
```

Comment:
I myself also wanted to do this but did not find a way without complicating the code too much.